src/v3/src/transformer/field/transform.ts

```diff
@@ -91,12 +91,18 @@ export const transformStepInputs = (
       const uischema = mapUiElement(input);
       acc.uischema.elements = [...acc.uischema.elements, uischema];
 
+      if (type === 'boolean' && required) {
+        acc.data[name] = Boolean(input.value);
+      }
```

Comment:
For required boolean inputs (checkboxes), we just set it as what is provided in value field, if there is nothing in value field, it will be false (unchecked) by default.

Reply:
> I'm not totally clear why we need to set the value explicitly this way, can you explain a bit more or give an example of where this is necessary?

In the case of _required_ boolean (checkboxes), we need to set the value in the data bag as a boolean. If we don't and the user never checks (or checks and unchecks) the box, the value will be `undefined` in the data bag and once the submission is made, it will fail (since the boolean is required). The change here is to set the field with a default boolean value to prevent that error early on. 

> Why do we need to use the Boolean function here?

I used boolean because `input.value` is not always set, it can be undefined. So the idea was to set the value to what is available in the input by default, or if a value isnt set, make it `false` by default. 

Additionally, the type of `input.value` is `string` (even though it is truly a boolean when the type is a boolean) so this sort of gets around that.

Reply:
Good point, in this scenario, if the type of the input is ever === `boolean` and the `value` is set, I have only ever seen it (the value) set as an actual boolean or undefined (either `true` or `false` or `undefined`). I have never seen a boolean type value set as a string, granted I have only seen the value set in one scenario (which is the OV push notification scenario). 

In the scenario of profile enrollment, I don't see a default value passed (and not sure if one can even be set from admin dashboard). 

Should I treat it as it it can be a string?

Reply:
Might be safer and avoid potential confusion in the future if we do an explicit check:
```suggestion
      if (type === 'boolean' && required) {
        acc.data[name] = input.value === true;
      }
```
I think this will work the same way as the `Boolean` way but also default to `false` if a string is passed as `input.value`

Reply:
I had to make the change this way because the type defined in auth-js does not account for booleans (even though it can be boolean and will be in this scenario).

```
if (type === 'boolean' && required) {
   acc.data[name] = (input.value as unknown as boolean) === true;
}
```

Reply:
Oh, does it not allow you to compare `unknown` with `boolean`? Seems like it would safe to allow this.

Reply:
Oh yea, I can just make it `unknown`, I will update that.